docs/docs/kb/12-plugins.md

```diff
@@ -0,0 +1,140 @@
+
+``` yaml
+plugins:
+    # must be the name same name as the plugin project name
+  - plugin:
+    # path the plugin project directory 
+    # path can be a remote git repository or local directory
+    path: /path/to/plugin
+    # parameters which can be passed to commands when executing
+    with:
+        foo: value
+        bar: value
+```
+
+## Plugin Commands
+Plugin commands are custom commands added to the ignite cli by a `registered` plugin. Commands can be of any path not defined already by ignite. All plugin commands will extend of the command root `ignite`. An example command definition is outlined below:
+```go
+func (p) Commands() []plugin.Command {
+	// TODO: write your command list here
+	return []plugin.Command{
+        {
+            Use:               "plugin",
+            Short:             "Explain what the command is doing...",
+            Long:              "Long description goes here...",
+            PlaceCommandUnder: "ignite chain",
+            // Examples of subcommands:
+            Commands: []plugin.Command{
+                {Use: "add"},
+                {Use: "list"},
+                {Use: "delete"},
+            },
+        },
+    }
+}
+
+func (p) Execute(cmd plugin.Command, args []string) error {
+	// TODO: write command execution here
+	fmt.Printf("Hello I'm the test-plugin plugin!\nargs=%v, with=%v\n", args, cmd.With)
+
+	// This is how the plugin can access the chain:
+	c, err := ignitecmd.NewChainWithHomeFlags(cmd.CobraCmd)
+	if err != nil {
+		return err
+	}
+	_ = c
+
+	// According to the number of declared commands, you may need a switch:
+	switch cmd.Use {
+	case "add":
+		fmt.Println("Adding stuff...")
+	case "list":
+		fmt.Println("Listing stuff...")
+	case "delete":
+		fmt.Println("Deleting stuff...")
+	}
+	return nil
+}
+```
+
+Above we see that the basic properties of the new `Command` are declared in the `Plugin.Command` struct which oulines the basic information of the new command. The `Use` property can be considered the `command root` of your custom functionality.
+
+Each time a command from a plugin is invoked, for example `ignite plugin add` it will invoke the `Execute` method declared within the plugin. You'll notice that mutiple branches are defined on the command. All custom behavor of your command should be implemented within `Execute`. All commands defined within the plugin will invoke `Execute`.
+
+## Plugin Hooks
+
+Plugin `Hooks` allow existing igntie commands to be extended with new functionality. Hooks are useful when you want to streamline functionality without needing to run custom scripts after or before a command has been run. this can streamline processes that where once error prone or forgotten all together.
```

Comment:
```suggestion
Plugin `Hooks` allow existing ignite commands to be extended with new functionality. Hooks are useful when you want to streamline functionality without needing to run custom scripts after or before a command has been run. this can streamline processes that where once error prone or forgotten all together.
```